Feature/selection features

kylejsmith12 opened the pull request.

kylejsmith12 committed:
Selection Feature

added the ability to hover over wires for so you know that those are selected using ray tracing.

kylejsmith12 committed:
Refactoring

followed tutorial to refactor code so that it is much more universal and stuff

kylejsmith12 committed:
Pickup the wirecutters

This does not work yet. Possibly have a script so that you can pick up and set down the wires? Who knows

C-Hess commented on unity/Assets/HighlightSelectionResponse.cs:
```diff
@@ -0,0 +1,7 @@
+﻿using UnityEngine;
+
+internal interface HighlightSelectionResponse
```
This is a duplicate of ISelectionResponse.

link46223 replied:
Is it something we need to look at? I think that the HighlightSelectionResponse should be fine?

C-Hess replied:
Why do we have duplicate interfaces with different names?

kylejsmith12 replied:
I was gonna add a feature but decided not to, kept the code just in case I did want to re-implement it

kylejsmith12 replied:
Actually shoot, this was from a previous attempt, this file shouldnt even be here. I think it was from when I was doing stuff in the master branch and then everything got messed up.

link46223 committed:
Merge branch 'master' into feature/Selection-features

kylejsmith12 committed:
Delete HighlightSelectionResponse.cs

link46223 committed:
Merge branch 'master' into feature/Selection-features

kylejsmith12 committed:
Added script

added script to spin the wirecutters, so that it is obvious to the user they need to be picked up.

kylejsmith12 committed:
Destroy wire when clicked

Script that destroys the wire when clicked to defuse the bomb
